src/webgpu/shader/execution/expression/call/builtin/subgroupMul.spec.ts

```diff
@@ -0,0 +1,374 @@
+export const description = `
+Execution tests for subgroupMul and subgroupExclusiveMul
+
+Note: There is a lack of portability for non-uniform execution so these tests
+restrict themselves to uniform control flow.
+Note: There is no guaranteed mapping between subgroup_invocation_id and
+local_invocation_index. Tests should avoid assuming there is.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf, objectsToRecord } from '../../../../../../common/util/data_tables.js';
+import { iterRange } from '../../../../../../common/util/util.js';
+import { GPUTest } from '../../../../../gpu_test.js';
+import {
+  kConcreteNumericScalarsAndVectors,
+  Type,
+  VectorType,
+  numberToFloatBits,
+  floatBitsToNumber,
+  kFloat32Format,
+  kFloat16Format,
+  scalarTypeOf,
+} from '../../../../../util/conversion.js';
+import { FP } from '../../../../../util/floating_point.js';
+
+import {
+  kNumCases,
+  kStride,
+  kWGSizes,
+  kPredicateCases,
+  runAccuracyTest,
+  runComputeTest,
+} from './subgroup_util.js';
+
+export const g = makeTestGroup(GPUTest);
+
+const kIdentity = 1;
+
+const kDataTypes = objectsToRecord(kConcreteNumericScalarsAndVectors);
+
+const kOperations = ['subgroupMul', 'subgroupExclusiveMul'] as const;
+
+g.test('fp_accuracy')
+  .desc(
+    `Tests the accuracy of floating-point multiplication.
+
+The order of operations is implementation defined, most threads are filled with
+the identity value and two receive random values.
+Subgroup sizes are not known ahead of time so some cases may not perform any
+interesting operations. The test biases towards checking subgroup sizes under 64.
+These tests only check two values in order to reuse more of the existing infrastructure
+and limit the number of permutations needed to calculate the final result.`
+  )
+  .params(u =>
+    u
+      .combine('case', [...iterRange(kNumCases, x => x)])
+      .combine('type', ['f32', 'f16'] as const)
+      .combine('wgSize', [
+        [kStride, 1, 1],
+        [kStride / 2, 2, 1],
+      ] as const)
+  )
+  .beforeAllSubcases(t => {
+    const features: GPUFeatureName[] = ['subgroups' as GPUFeatureName];
+    if (t.params.type === 'f16') {
+      features.push('shader-f16');
+      features.push('subgroups-f16' as GPUFeatureName);
+    }
+    t.selectDeviceOrSkipTestCase(features);
+  })
+  .fn(async t => {
+    await runAccuracyTest(
+      t,
+      t.params.case,
+      [t.params.wgSize[0], t.params.wgSize[1], t.params.wgSize[2]],
+      'subgroupMul',
+      t.params.type,
+      kIdentity,
+      t.params.type === 'f16' ? FP.f16.multiplicationInterval : FP.f32.multiplicationInterval
+    );
+  });
+
+/**
+ * Checks subgroup multiplications.
+ *
+ * Expected results:
+ * - subgroupMul: each invocation should have result equal to real subgroup size
```

Comment:
This comment is copypasta from the addition case.

The code is checking against powers-of-two.